src/gpgmm/common/CPUPoolAllocator.h

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 The GPGMM Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GPGMM_COMMON_CPUPOOLALLOCATOR_H_
+#define GPGMM_COMMON_CPUPOOLALLOCATOR_H_
+
+#include "gpgmm/utils/Assert.h"
+
+#include <queue>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+using AllocationAddress = std::pair<uint64_t, uint64_t>;
+
+namespace gpgmm {
+    template <typename T>
+    class CPUPoolAllocator {
```

Comment:
nit: good to have a "explainer" (how to use, Q&A, etc).

Reply:
Done.